doc/clients.md

```diff
@@ -0,0 +1,96 @@
+# Jackdaw Client API
+
+## Rationale
+
+The Jackdaw Client API wraps the core Kafka `Producer`<sup>[1](#producerapi)</sup> and
+`Consumer`<sup>[2](#consumerapi)</sup> APIs and provides functions for building or
+unpacking some of the supporting objects like Callbacks, Serdes, ConsumerRecords etc.
+
+## Producing
+
+The producer example below demonstrates how to use the Kafka Producer API. The configuration<sup>[3](#producerconfig)</sup>
+is represented as a simple map (Jackdaw will convert this to a `Properties` object), and the
+key and value serializers are specified when creating the producer to override the default
+(which would be the `StringSerializer`).
+
+Producers are usually created using the `with-open` macro so that they are automatically
+closed either when evaluation reaches the end of the body or an exception is thrown.
+
+Within the body, the `jc/produce!` function is used to request a write to the specified
+Kafka topic. This function returns a delay immediately which can be `deref`'d to wait
+for the result of the Kafka `.send` call which includes metadata like the timestamp
+and offset of the written record.
+
+In this example, the JSON serde converts the message from a plain old Clojure map into a
+JSON byte-array while building the ProducerRecord that is eventually passed to the
+KafkaProducer's `.send` method. Other Serdes are available<sup>[4](#serdesdirectory)</sup>
+
+```
+(ns producer-example
+  (:require
+    [jackdaw.serdes.json :as json]
```

Comment:
If we use a string serializer for the value, we can remove this dependency.

Reply:
Yeah that's probably a good idea. I think it will be good to describe somewhere how serdes can be used to automatically marshall plain old clojure data into whatever serialization format is desired but that probably merits its own doc.